cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -100,6 +102,32 @@ func (w *Watcher) RecordUsage(e entry) {
 	w.an.EventWithSource(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, e.Source, e.Dims)
 }
 
+// GetProcessesInUse returns for a given runtime exec directory a map of running executables to
+// their process IDs.
+func (w *Watcher) GetProcessesInUse(execDir string) map[string]int {
+	inUse := map[string]int{}
+
+	if runtime.GOOS != "linux" {
+		execDir = strings.ToLower(execDir) // Windows and macOS filesystems are case-insensitive
+	}
+	for _, proc := range w.watching {
+		exeToCompare := proc.Exec
+		if runtime.GOOS != "linux" {
+			exeToCompare = strings.ToLower(exeToCompare) // Windows and macOS filesystems are case-insensitive
+		}
+		if !strings.Contains(exeToCompare, execDir) {
+			continue
+		}
+		if isRunning, err := proc.IsRunning(); err == nil && isRunning {
```

Comment:
I've removed this check because by definition, if the process is being watched, it's currently in use. Once it's no longer in use, it will not be in the watch list and will not end up being in the returned list.

Reply:
Yeah, the only downside here is that the watcher triggers on a 1 minute interval.. And given how slow process watching is I'm concerned that making that interval shorter would have performance impacts.

This means that if someone runs into this error, and closes the offending process, they may have to wait a minute before they can retry..

I think it's still worth having your original check, so long as we can confirm that it doesn't slow the user down in any noticeable way.

Reply:
It had the same impact as the "awkward pause" noted in https://activestatef.atlassian.net/browse/DX-2472. If we added the check back, we're just putting the awkward pause in the service. The state tool user will still see it when waiting for a response from the service.

Reply:
I thought about this overnight and since we're only invoking macOS's `lsof` for each matching process (instead of all system processes), there shouldn't be a slowdown. I just tested with debug log timestamps and confirmed that there isn't a slowdown like I thought there would be yesterday:

```
(PID 12284) [DBG 12:02:52.218 entry.go:26] Checking if /Users/me/Library/Caches/activestate/de3cdb9a/exec/python (12294) is still running
(PID 12284) [DBG 12:02:52.295 watcher.go:119] Runtime process 12294:/Users/me/Library/Caches/activestate/de3cdb9a/exec/python is not running
```